plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -124,7 +124,6 @@ public class IcebergSplitSource
     private long targetSplitSize;
     private CloseableIterator<FileScanTask> fileScanIterator;
     private Iterator<FileScanTask> fileTasksIterator = emptyIterator();
-    private boolean fileHasAnyDeletions;
```

Comment:
> Refactor IcebergSplitSource to reduce per split operations

the diff looks big, but i guess its because of some code reordering. how should i read it best?

Reply:
Mostly this is lifting the per-split logic and moving it to per-file. There is small change to where the limit handling is applied.
I would just look at the end result and see if that makes sense.